Source/Scene/TextureAtlas.js

```diff
@@ -74,6 +74,7 @@ function TextureAtlas(options) {
   this._textureCoordinates = [];
   this._guid = createGuid();
   this._idHash = {};
+  this._indexHash = {};
```

Comment:
It's a little unfortunate that we need to have two objects keeping track here: `id` -> `indexHash[id]` but I suppose we just need the `TextureAtlas` operates with the `index`. Is that correct?

Reply:
> but I suppose we just need the TextureAtlas operates with the index

I'm not sure exactly what you mean. But in the case where we are asynchronously adding an image, we do need to be able to return the `Promise`, not just the resulting `index`.